README.md

```diff
@@ -3,11 +3,221 @@
+
+```tsx
+import { useInfiniteQuery } from '@tanstack/react-query';
+import { widgetsInfiniteQueryOptions } from './api/hooks/widgets';
+
+function InfiniteWidgetList() {
+  const { data, fetchNextPage, hasNextPage } = useInfiniteQuery(
+    widgetsInfiniteQueryOptions(),
+  );
+
+  return (
+    <>
+      {data?.pages.map((page) =>
+        page.edges.map((widget) => <div key={widget.id}>{widget.name}</div>),
+      )}
+      {hasNextPage && (
+        <button onClick={() => fetchNextPage()}>Load More</button>
+      )}
+    </>
+  );
+}
+```
+
+## Configuration Options
+
+The generator accepts the following options in your `basketry.config.json`:
+
+```json
+{
+  "generators": [
+    {
+      "rule": "@basketry/react-query",
+      "options": {
+        "reactQuery": {
+          "typesModule": "../types",
+          "clientModule": "../http-client",
+          "typeImports": true,
+          "includeVersion": true,
+          "eslintDisable": ["no-unused-vars"],
+          "prettierConfig": ".prettierrc.json"
+        }
+      }
+    }
+  ]
+}
+```
+
+## Query Key Structure
+
+The generator creates consistent query keys following this pattern:
+
+```typescript
+[serviceName, methodName, params || {}] as const;
+```
+
+This structure enables:
+
+- Hierarchical cache invalidation (e.g., invalidate all widget queries)
+- Precise cache updates for specific queries
+- Better debugging with readable query keys
+
+### Type-Safe Query Key Builder
+
+Starting with v0.2.0, the generator also creates a `matchQueryKey` function that provides type-safe query key building for cache operations:
+
+```tsx
+import { matchQueryKey } from './api/hooks/query-key-builder';
+import { useQueryClient } from '@tanstack/react-query';
+
+function MyComponent() {
+  const queryClient = useQueryClient();
+
+  // Invalidate all queries for a service
+  queryClient.invalidateQueries({
+    queryKey: matchQueryKey('widget'),
+  });
+
+  // Invalidate all queries for a specific operation
+  queryClient.invalidateQueries({
+    queryKey: matchQueryKey('widget', 'getWidgets'),
+  });
+
+  // Invalidate a specific query with parameters
+  queryClient.invalidateQueries({
+    queryKey: matchQueryKey('widget', 'getWidgets', { status: 'active' }),
+  });
+
+  // Type-safe with IntelliSense support
+  // TypeScript will autocomplete service names, operations, and validate param types
+}
+```
+
+The `matchQueryKey` function provides:
+
+- **Type Safety**: Full TypeScript support with autocomplete for services, operations, and parameters
+- **Partial Matching**: Build keys at any granularity level for flexible cache invalidation
+- **Consistency**: Guarantees keys match the structure used by generated query options
+- **Developer Experience**: IntelliSense guides you through available options
+
+## Migration from v0.1.x
+
+If you're upgrading from v0.1.x (wrapper hooks pattern) to v0.2.0 (queryOptions pattern), see the [Migration Guide](./MIGRATION.md)
```

Comment:
Let's not include this in the initial PR. Instead, let's add these type of docs into the `basketry.github.io` repo.